packages/charts/src/charts/table/row.component.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Ceil } from '@keen.io/ui-core';
+import { StripedTable } from './table.styles';
+import { copyToClipboard } from './table.utils';
+import { FormatFuncType } from '../../types';
+
+type Props = {
+  data?: any;
+  color: string;
+  format?: FormatFuncType;
+};
+
+export const Row = ({ data, color, format }: Props) => {
+  return (
+    <StripedTable color={color}>
+      {Object.keys(data).map(key => (
+        <Ceil
+          key={`${key}`}
+          onClick={copyToClipboard}
+          format={typeof format === 'object' ? format[key] : format}
```

Comment:
The same here - if we move data preparation to higher layer (`generateTable`) at root component level. The `Row` will be only presentational and not dependent on `format` function.